api/src/controllers/reviews.ts

```diff
@@ -181,6 +192,35 @@ router.patch("/vote", async function (req, res) {
   }
 });
 
+/**
+ * Get whether or not the color of a button should be colored
+ */
+router.patch("/getVoteColor", async function (req, res) {
```

Comment:
I would make this return a dictionary that provides the votes of a given user (map reviewID to vote status boolean). That way we don't need to make a request for each subreview. Only 1 request will be needed per page to see all vote colors. Could possibly take in a list of reviewIDs too so we don't relay irrelevant reviews.